src/core/toga/widgets_registry.py

```diff
@@ -0,0 +1,32 @@
+class WidgetsRegistry:
```

Comment:
Linguistic nit - `WidgetRegistry` would make make more sense.

Also - is there any reason that this shouldn't be a subclass of dict, rather than a proxy wrapper? That would significantly simplify the implementation, AFAICT.

Reply:
I changed the name to `WidgetRegistry`.

I did not want to inherit from `dict` because I did not want anyone to use `__setitem__` directly. since users will use this class directly via `App.widget` and `Window.widget`.

I can do that, but I'm not sure it's the best idea.

Reply:
Sure - but you can then override the one method you *don't* want people to use, and have it raise an exception.

Reply:
Alternatively - if we're not trying to preserve a "dict" API (as the use of `add`, `extend` and `remove` rather than `__setitem__`, `update` and `__del__` would possibly suggest), then we should remove the APIs that make the registry class appear "dict-like". Is there any use case for `__len__`, `__iter__` or `__repr__`, other than testing?

If we go down this path, I'd also be inclined to lean *away* from collection-like naming - i.e., use `register()` and `deregister()` rather than `add()` and `remove()`. 

That said - we've generally tried to maintain Pythonic APIs, so I'm not sure this is the right approach.